include/core/CFloatStorage.h

```diff
@@ -93,40 +93,34 @@ class CORE_EXPORT CFloatStorage {
     //! \name Operators
     //@{
     CFloatStorage operator-() const { return CFloatStorage{-m_Value}; }
-    bool operator==(const CFloatStorage& rhs) const {
-        return m_Value == rhs.m_Value;
-    }
-    bool operator==(const double& rhs) const {
+    bool operator==(CFloatStorage rhs) const { return m_Value == rhs.m_Value; }
```

Comment:
Why do you remove const reference here?

Reply:
For built in types it is actually better not to pass by reference. For example, this would require indirection when accessing rhs.m_Value and there is no overhead from the copy. Of course the compiler may elide this altogether by the as-if rule: copying would have no side-effects relative to pass by reference. However, I figured it is better not to rely on this behaviour.